src/scenic/core/object_types.py

```diff
@@ -283,6 +284,28 @@ class Point(_Constructible):
 	def visibleRegion(self):
 		return CircularRegion(self.position, self.visibleDistance)
 
+	def getModifiedVisibilityRegion(self, other, actors):
+		modifiedVisibilityRegion = self.visibleRegion
+		for actor in actors:
```

Comment:
Here, instead of iterating through all actors, can we first check which actors are in the self.visibleRegion, and then do the difference with only those? This will likely save time

Reply:
I do not believe it would save time, but I need to change this logic in the future anyway. We can discuss this further when the logic for the blocking cone being partially contained in the visible region is implemented.